File: apollo-ws/reference-implementations/python/trunk/ApolloFactory.py
# Copyright 2012 University of Pittsburgh
#
# Licensed under the Apache License, Version 2.0 (the "License"); you may not
# use this file except in compliance with the License.  You may obtain a copy of
# the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
# License for the specific language governing permissions and limitations under
# the License.


'''
Created on Nov 29, 2012

@author: John Levander
'''

from EpidemicSimulatorService_client import *
from EpidemicSimulatorService_types import *

class ApolloFactory:
    
    def new_epidemic_model_input(self):
        em_input = ns1.epidemic_model_input_Def(None).pyclass()
        em_input._authentication = ns1.authentication_Def(None).pyclass()
        em_input._simulator_identification = ns1.simulator_identification_Def(None).pyclass()
        em_input._simulator_configuration = ns1.simulator_configuration_Def(None).pyclass()
        em_input._disease_dynamics = ns1.disease_Def(None).pyclass()
        em_input._antiviral_control_measure = ns1.antiviral_control_measure_Def(None).pyclass()
        em_input._vaccination_control_measure = ns1.vaccination_control_measure_Def(None).pyclass()
        return em_input
    
    def new_epidemic_model_output(self):
        epidemic_model_output = ns1.epidemic_model_output_Def(None).pyclass()
#       self._population_time_series = None
#       self._control_measure_time_series = None

        epidemic_model_output._population_time_series = ns1.population_time_series_Def(None).pyclass()
#       self._pop_count = []
#       self._simulated_population = []

        epidemic_model_output._population_time_series._pop_count = [0.0] * 365
        
        epidemic_model_output._population_time_series._simulated_population = []
        
        epidemic_model_output._control_measure_time_series = ns1.control_measure_time_series_output_Def(None).pyclass()
#       self._received_vaccination = None
#       self._received_antiviral_treatment = None
        epidemic_model_output._control_measure_time_series._received_vaccination = []
        epidemic_model_output._control_measure_time_series._received_antiviral_treatment = []
        
        return epidemic_model_output
    
    def create_run_result(self):
        run_result = ns1.simulation_run_result_Def(None).pyclass()
#       self._model_output = None
#       self._runId = None
        run_result._runId = -1
        return run_result 
    
    def create_run_status(self):
        run_status = ns1.run_status_Def(None).pyclass()
        return run_status
        
        
        

File: apollo-ws/schemas/tags/0.1.0/src/main/java/edu/pitt/apollo/types/ApolloFactory.java

/* Copyright 2012 University of Pittsburgh
*
* Licensed under the Apache License, Version 2.0 (the "License"); you may not
* use this file except in compliance with the License.  You may obtain a copy of
* the License at
*
*   http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
* License for the specific language governing permissions and limitations under
* the License.
*/

package edu.pitt.apollo.types;

import java.util.ArrayList;

public class ApolloFactory {
	public static EpidemicModelInput createEpidemicModelInput() {
		EpidemicModelInput input = new EpidemicModelInput();
		input.setAntiviralControlMeasure(createAntiviralControlMeasure());
		input.setVaccinationControlMeasure(createVaccinationControlMeasure());
		input.setDiseaseDynamics(createDisease());
		input.setSimulatorIdentification(new SimulatorIdentification());
		input.setSimulatorConfiguration(new SimulatorConfiguration());
		input.setAuthentication(new Authentication());
		return input;
	}

	public static EpidemicModelOutput createEpidemicModelOutput() {
		EpidemicModelOutput output = new EpidemicModelOutput();
		output.controlMeasureTimeSeries = new ControlMeasureTimeSeriesOutput();
		output.controlMeasureTimeSeries.receivedAntiviralTreatmentTimeSeries = new ArrayList<Double>();
		output.controlMeasureTimeSeries.receivedVaccinationTimeSeries = new ArrayList<Double>();
		output.populationTimeSeries = new PopulationTimeSeries();
		output.populationTimeSeries.popCount = new ArrayList<Double>();
		output.populationTimeSeries.simulatedPopulation = new ArrayList<String>();
		return output;
	}

	public static Disease createDisease() {
		Disease disease = new Disease();
		disease.popCount = new ArrayList<Double>();
		disease.simulatedPopulation = new ArrayList<String>();
		return disease;
	}

	public static AntiviralControlMeasure createAntiviralControlMeasure() {
		AntiviralControlMeasure acm = new AntiviralControlMeasure();
		acm.antiviralAdminSchedule = new ArrayList<Double>();
		acm.antiviralSupplySchedule = new ArrayList<Double>();
		return acm;
	}

	public static VaccinationControlMeasure createVaccinationControlMeasure() {
		VaccinationControlMeasure vcm = new VaccinationControlMeasure();
		vcm.vaccinationAdminSchedule = new ArrayList<Double>();
		vcm.vaccineSupplySchedule = new ArrayList<Double>();
		return vcm;
	}
}
